src/mpl/configure.ac

```diff
@@ -856,10 +856,47 @@ if test "$ac_cv_func_usleep" = "yes" ; then
 fi
 
 AC_CHECK_FUNCS(backtrace_symbols)
-AC_CHECK_LIB(backtrace, backtrace_create_state)
-AC_CHECK_LIB(unwind, unw_backtrace)
 
-AC_CHECK_DECLS([backtrace_create_state, backtrace_print])
+# Check for libbacktrace support
+PAC_PUSH_FLAG([LIBS])
+LIBS="-lbacktrace"
+AC_LINK_IFELSE([
+    AC_LANG_PROGRAM([
+#include <backtrace.h>
+],[
+backtrace_create_state(0, 0, 0, 0);
+backtrace_print(0, 0, 0);
+])],
+    [ac_cv_lib_backtrace=yes],
+    [ac_cv_lib_backtrace=no]
+)
+PAC_POP_FLAG([LIBS])
+
+if test "$ac_cv_lib_backtrace" = "yes" ; then
+    AC_DEFINE([HAVE_BACKTRACE_H],[1],[Define to 1 if you have the <backtrace.h> header file.])
+    AC_DEFINE([HAVE_LIBBACKTRACE],[1],[Define to 1 if you have the backtrace library (-lbacktrace)])
+    PAC_PREPEND_FLAG([-lbacktrace],[LIBS])
+fi
+
+# Check for libunwind support
+PAC_PUSH_FLAG([LIBS])
+LIBS="-lunwind"
+AC_LINK_IFELSE([
+    AC_LANG_PROGRAM([
+#include<libunwind.h>
+],[
+unw_backtrace(0, 0);
+])],
+    [ac_cv_lib_libunwind=yes],
+    [ac_cv_lib_libunwind=no]
+)
+PAC_POP_FLAG([LIBS])
+
+if test "$ac_cv_lib_libunwind" = "yes" ; then
+    AC_DEFINE([HAVE_LIBUNWIND_H],[1],[Define to 1 if you have the <libunwind.h> header file.])
+    AC_DEFINE([HAVE_LIBUNWIND],[1],[Define to 1 if you have the libunwind library (-lunwind)])
```

Comment:
What's the point of having both these defines? Since they are either both set or not set, can they be combined into a single define?

Reply:
Yes that would make more sense. Initially there were two separate checks for headers and libs, now that there is only one this can just become HAVE_LIBUNWIND. I will reflect this in the MPL code as well then.